src/components/Input/Input.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { StyleSheet, Text, TextInput, View } from 'react-native';
+
+import { useTheme } from '@/hooks';
+
+type Props = React.ComponentPropsWithRef<typeof TextInput> & {
+  label: string;
+  secureTextEntry?: boolean;
+  onSubmit: (text: string) => void;
+};
+
+const Input = ({ ...props }: Props) => {
+  const { themedStyles } = useTheme();
+  return (
+    <View style={styles.container}>
+      <Text style={{ ...styles.label, color: themedStyles.text }}>
+        {props.label}
+      </Text>
+      <TextInput
+        style={{
+          ...styles.input,
+          backgroundColor: themedStyles.inputBackground,
+          color: themedStyles.text,
+          borderColor: themedStyles.inputBorder,
+        }}
+        defaultValue=""
```

Comment:
chyba nie ma potrzeby nadawania wartości domyślnej w tym przypadku, gdy nie przekażemy nic, to będzie undefined, co będzie łatwiej zwalidować